src/CryptoHelper.php

```diff
@@ -86,14 +95,28 @@ public static function sign($data, $cert, $privateKey = null, $headers = [], $mi
     }
 
     /**
-     * @param string|MimePart $data
-     * @param array|null      $caInfo    Information about the trusted CA certificates to use in the verification process
-     * @param array           $rootCerts
+     * @param string|MimePart   $data
+     * @param array|null        $caInfo    Information about the trusted CA certificates to use in the verification process
+     * @param array             $rootCerts
+     * @param bool              $encodeData
      *
      * @return bool
      */
-    public static function verify($data, $caInfo = null, $rootCerts = null)
+    public static function verify($data, $caInfo = null, $rootCerts = null, $encodeData = false)
     {
+        if ($encodeData === true) {
+            $temp = new MimePart($data->getHeaders());
+            foreach ($data->getParts() as $part) {
+                if (self::stringIsBinary($part->getBody())) {
+                    $encodedPart = new MimePart($part->getHeaders(), Utils::encodeBase64($part->getBody()));
+                    $temp->addPart($encodedPart);
+                } else {
+                    $temp->addPart($part);
+                }
+            }
+            $data = $temp;
```

Comment:
Wouldn't it be better to extract this to an "encodePayload" method?

Reply:
👍